### ISSUE TITLE
Multiple networking module regression?

### ISSUE THREAD

**Issue by minijackson** [Alexays/Waybar#246]
Hello! I recently upgraded Waybar from 0.4.0 to 0.5.0, and I have two networking modules enabled (one for ethernet, one for wireless) and two monitors. With waybar-0.4.0 everything was fine but with waybar-0.5.0 I can only see the ethernet module one monitor (probably because the ethernet is the first in the config).

The console log says:
```
Disabling module "network#wlan", Can't add membership
Disabling module "network#eth", Can't add membership
Disabling module "network#wlan", Can't add membership
```

and a quick strace shows a `bind` syscall which returns "Address already in use" just before each errors.

A minimal config reproducing the issue (even with only 1 monitor):

```json
{
  "modules-right": ["network#1", "network#2", "clock"],
  "network#1": { "interface": "enp3s0" },
  "network#2": { "interface": "enp3s0" }
}
```

Thanks for the great work!

**Comment by Alexays** [comment]
Can you test if with the `network` branch your issue is resolved?
(949a4ecf2e4619bd5eb74c84bccc0e06e79f9594)

**Comment by minijackson** [comment]
So I've tested both the commit 949a4ecf2e4619bd5eb74c84bccc0e06e79f9594 and the `network` branch (which points to 25379f3bc9159a42ecf2f7b32dc4d084445e66cb instead of 949a4ecf2e4619bd5eb74c84bccc0e06e79f9594).

Unfortunately, the `network` branch doesn't show any network module anymore (`Can't add membership`) and the 949a4ecf2e4619bd5eb74c84bccc0e06e79f9594 commit doesn't compile:

```
FAILED: waybar@exe/src_modules_network.cpp.o
<Very long compilation command line of `../src/modules/network.cpp` due to NixOS>
In file included from /<...>-libnl-3.4.0-dev/include/libnl3/netlink/netlink.h:25:0,
                 from ../include/modules/network.hpp:6,
                 from ../src/modules/network.cpp:2:
../src/modules/network.cpp: In static member function 'static int waybar::modules::Network::handleEvents(nl_msg*, void*)':
../src/modules/network.cpp:402:52: error: 'ret' was not declared in this scope
   for (nlmsghdr *nh = nlmsg_hdr(msg); NLMSG_OK(nh, ret);
                                                    ^
../src/modules/network.cpp:402:52: note: suggested alternative: 'net'
../src/modules/network.cpp:415:11: error: return-statement with no value, in function returning 'int' [-fpermissive]
           return;
           ^~~~~~
```

**Comment by minijackson** [comment]
Here's the `strace` output just before a "Can't add membership error message", if it helps:

```
socket(AF_NETLINK, SOCK_RAW|SOCK_CLOEXEC, NETLINK_ROUTE) = 34
setsockopt(34, SOL_SOCKET, SO_SNDBUF, [32768], 4) = 0
setsockopt(34, SOL_SOCKET, SO_RCVBUF, [32768], 4) = 0
getpid()                                = 30859
bind(34, {sa_family=AF_NETLINK, nl_pid=-1455392629, nl_groups=00000000}, 12) = 0
getsockname(34, {sa_family=AF_NETLINK, nl_pid=-1455392629, nl_groups=00000000}, [12]) = 0
setsockopt(34, SOL_NETLINK, NETLINK_ADD_MEMBERSHIP, [1], 4) = 0
setsockopt(34, SOL_NETLINK, NETLINK_ADD_MEMBERSHIP, [16], 4) = 0
setsockopt(34, SOL_NETLINK, NETLINK_ADD_MEMBERSHIP, [64], 4) = -1 EINVAL (Invalid argument)
poll([{fd=9, events=POLLIN}], 1, 0)     = 1 ([{fd=9, revents=POLLIN}])
write(2, "Disabling module \"network#eth\", "..., 52Disabling module "network#eth", Can't add membership) = 52
```

**Comment by Alexays** [comment]
Oh ok, i think i got it, it might be fixed on master ;)

**Comment by minijackson** [comment]
Unfortunately I just tested 7fac48353046a6ed71bd7f96f2b67d83ca7949ff and it's back to the way I first described. Here's the corresponding strace output:

```c
socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE) = 34
bind(34, {sa_family=AF_NETLINK, nl_pid=32764, nl_groups=0x000111}, 12) = -1 EADDRINUSE (Address already in use)
poll([{fd=8, events=POLLIN}], 1, 0)     = 1 ([{fd=8, revents=POLLIN}])
write(2, "Disabling module \"network#eth\", "..., 52) = 52
```

**Comment by Alexays** [comment]
Just reverted to 0.4.0 state, i'll find another fix for #156
